sbtc/src/rpc.rs

```diff
@@ -37,10 +42,179 @@ pub struct GetTxResponse {
+    pub confirmations: u32,
+    /// The Unix epoch time when the block was mined. It reflects the
+    /// timestamp as recorded by the miner of the block.
+    #[serde(rename = "blocktime")]
+    pub block_time: u64,
+}
+
+impl BitcoinTxInfo {
+    /// Assess how much of the bitcoin miner fee should be apportioned to
+    /// the input associated with the given `outpoint`.
+    ///
+    /// # Notes
+    ///
+    /// Each input and output is assessed a fee that is proportional to
+    /// their weight amount all the requests serviced by this
+    /// transaction.
+    ///
+    /// This function assumes that this transaction is an sBTC transaction,
+    /// which implies that the first input and the first two outputs are
+    /// always the signers'. So `None` is returned if there is no input,
+    /// after the first input, with the given `outpoint`.
+    pub fn assess_input_fee(&self, outpoint: OutPoint) -> Option<Amount> {
+        // The Weight::to_wu function just returns the inner weight units
+        // as an u64, so this is really just the weight.
+        let request_weight = self.request_weight().to_wu();
+        // We skip the first input because that is always the signers'
+        // input UTXO.
+        let input_weight = self
+            .tx
+            .input
+            .iter()
+            .skip(1)
+            .find(|tx_in| tx_in.previous_output == outpoint)?
+            .segwit_weight()
+            .to_wu();
+
+        let fee_sats = (input_weight * self.fee.to_sat()).div_ceil(request_weight);
+        Some(Amount::from_sat(fee_sats))
+    }
+
+    /// Assess how much of the bitcoin miner fee should be apportioned to
+    /// the output at the given output index `vout`.
+    ///
+    /// # Notes
+    ///
+    /// Each input and output is assessed a fee that is proportional to
+    /// their weight amount all the requests serviced by this
+    /// transaction.
+    ///
+    /// This function assumes that this transaction is an sBTC transaction,
+    /// which implies that the first input and the first two outputs are
+    /// always the signers'. So `None` is returned if the given `vout` is 0
+    /// or 1 or if there is no output in the transaction at `vout`.
+    pub fn assess_output_fee(&self, vout: usize) -> Option<Amount> {
+        // We skip the first input because that is always the signers'
+        // input UTXO.
+        if vout < 2 {
+            return None;
+        }
+        let request_weight = self.request_weight().to_wu();
+        let input_weight = self.tx.output.get(vout)?.weight().to_wu();
+
+        let fee_sats = (input_weight * self.fee.to_sat()).div_ceil(request_weight);
+        Some(Amount::from_sat(fee_sats))
+    }
+
+    /// Computes the total weight of the inputs and the outputs, excluding
+    /// the ones related to the signers.
+    fn request_weight(&self) -> Weight {
+        // We skip the first input and output because those are always the
```

Comment:
I will try to always make this the case. Right now a lot hinges on this always being the case. The only way to know for sure is to check the `scriptPubKey` of the first input and first two outputs.

I'd like to move this closer to the code that depends on this always being the case, so that if we change it there it becomes more difficult to miss the need to change this but I didn't want to bloat the PR with a refactor.

Reply:
Gotcha -- should that be todo(#)'d or do you want to leave it?

Reply:
Comment updated in https://github.com/stacks-network/sbtc/pull/557/commits/e7d8dd96e2345ae0af01cb96c9c6a25a4d035828.

Reply:
> Gotcha -- should that be todo(#)'d or do you want to leave it?

I'd like to leave it for now.